src/Core/src/AbstractApi.php

```diff
@@ -144,12 +144,21 @@ protected function getSignerFactories(): array
     }
 
     /**
-     * @psalm-suppress InvalidNullableReturnType
+     * This method is a BC layer for client that does not require core:^1.2.
+     *
+     * @return array{endpoint: string, signRegion: string, signService: string, signVersions: string[]}
      */
-    protected function getEndpointPattern(?string $region): string
+    protected function getEndpointMetadata(string $region): array
     {
-        /** @psalm-suppress NullableReturnStatement */
-        return $this->configuration->get('endpoint');
+        /** @var string $endpoint */
+        $endpoint = $this->configuration->get('endpoint');
+
+        return [
+            'endpoint' => $endpoint,
+            'signRegion' => $region,
+            'signService' => $this->getSignatureScopeName(),
+            'signVersions' => [$this->getSignatureVersion()],
```

Comment:
We should deprecate these functions in SQS, S3 and SES. And then configure the generator to not include them in other clients. But that should be done in another PR.

Reply:
agreed